core/arch/arm/plat-marvell/main.c

```diff
@@ -108,3 +114,38 @@ void console_init(void)
 #endif
 	register_serial_console(&console_data.chip);
 }
+
+#ifdef CFG_HW_UNQ_KEY_SUPPORT
+TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
+{
+	uint8_t hw_unq_key[sizeof(hwkey->data)] = { };
+	void *fuse_huk1 = NULL;
+	void *fuse_huk2 = NULL;
+	uint64_t huk1 = 0;
+	uint64_t huk2 = 0;
+	uint8_t byte = 0;
+
+	fuse_huk1 = phys_to_virt(PLAT_MARVELL_FUSF_FUSE_BASE +
+							PLAT_MARVELL_FUSF_HUK_OFFSET,
+							MEM_AREA_IO_SEC);
+	fuse_huk2 = phys_to_virt(PLAT_MARVELL_FUSF_FUSE_BASE +
+							PLAT_MARVELL_FUSF_HUK_OFFSET + 8,
+							MEM_AREA_IO_SEC);
+	if (!fuse_huk1 || !fuse_huk2) {
+		EMSG("\nH/W Unique key is not fetched from the platform.");
+		return TEE_ERROR_SECURITY;
+	}
+
+	huk1 = get_le64(fuse_huk1);
+	huk2 = get_le64(fuse_huk2);
+
+	for (byte = 0; byte <= 7; byte++) {
+		hw_unq_key[byte] = (uint8_t)(((huk1 >> (byte * 8))) & 0xFF);
+		hw_unq_key[8 + byte] = (uint8_t)(((huk2 >> (byte * 8))) & 0xFF);
+	}
+
+	memcpy(&hwkey->data[0], hw_unq_key, sizeof(hwkey->data));
```

Comment:
Sorry one more comment. Why not just do: 
```C
memcpy(hwkey->data, fuse_huk1, sizeof(hwkey->data));
```
?

Reply:
> Sorry one more comment. Why not just do:
> 
> ```c
> memcpy(hwkey->data, fuse_huk1, sizeof(hwkey->data));
> ```
> 
> ?

As huk1 and huk2 fuse addresses are contiguous yes we can. Thanks for the suggestion.

Reply:
@anilreddy3 OK then please simplify the function as much as possible.